options.go

```diff
@@ -88,20 +87,18 @@ func Resolve(options interface{}, flagSet *flag.FlagSet, cfg map[string]interfac
 		// 3. config file option
 		// 4. default flag value
 		var v interface{}
-		if hasArg(flagName) {
+		if hasArg(flagSet, flagName) {
 			v = flagInst.Value.String()
-		} else if deprecatedFlagName != "" && hasArg(deprecatedFlagName) {
+		} else if deprecatedFlagName != "" && hasArg(flagSet, deprecatedFlagName) {
 			v = deprecatedFlag.Value.String()
 			log.Printf("WARNING: use of the --%s command line flag is deprecated (use --%s)",
 				deprecatedFlagName, flagName)
 		} else if cfgVal, ok := cfg[cfgName]; ok {
 			v = cfgVal
 		} else {
 			// if no flag arg or config file option was specified just use the default
-			// flag value
-			if getter, ok := flagInst.Value.(flag.Getter); ok {
-				v = getter.Get()
-			}
```

Comment:
I think this case is still required e.g. https://github.com/nsqio/nsq/blob/master/apps/nsqd/nsqd.go#L41

Perhaps this should be:

```go
if getter, ok := flagInst.Value.(flag.Getter); ok {
    v = getter.Get()
} else {
    v = val.Field(i).Interface()
}
```